on/off switch for running in the opposite direction (backwards)

Alexgamer470 opened the pull request:
I've added the on/off switch for running in the opposite direction.
For more information head to the latest commit in this branch.

staxfur committed:
added nice looking watermark

staxfur committed:
the field creation is now animated!

staxfur committed:
the version ist now shown

staxfur committed:
moved movement loop skip

Alexgamer470 committed:
downgraded CMake Version 

I've downgraded the minimum required CMake Version, so I can develope on Windows with WSL Ubuntu.

Alexgamer470 committed:
on/off switch for running in the opposite direction

by setting 'bool noOppDir = false' in file 'snake.h' you can run in the opposite direction as long as your score is equal to 0. If your score is greater than 0 you'll die from running in the opposite direction. Setting 'bool noOppDir' to 'false' prevents you from running in the opposite direction, so you can't die.

staxfur commented on src/snake.cpp:
```diff
@@ -417,19 +417,31 @@ void snake::snake::getInput()
                 case KEY_DOWN:
                 case 's':
-                    input = moveDown;
+                    if (!snake::snake::noOppDir)
```
namespace and class identifiers are not needed, if the function is inside the same class / namespace. You can just do !noOppdir

staxfur commented on src/snake.h:
```diff
@@ -53,6 +53,8 @@ namespace snake
         unsigned short int update();
         // destructor destroys game and ncurses
         virtual ~snake();
+        // if true you can't die from running in the opposite direction. bcs you can't run in the opposite direction.
```
Comments are not for telling someone your whole life xD, something like "toogles possible movement into the opposite direction" is enaugh.

staxfur reviewed (changes requested):
Good feature, some things should be changed before merging to make it more clean. Thank you for your work!

Cheers!

staxfur commented on src/snake.cpp:
```diff
@@ -417,19 +417,31 @@ void snake::snake::getInput()
-                    input = moveUp;
+                    if (!snake::snake::noOppDir)
+                        input = moveUp;
+                    if (input != moveDown)
```
lol just noticede, that this doesnt make any sense. just do if (!noOppDir || noOppdir && input != moveDown). And noOppDir is a really bad name, probably change that into something diffrent, like "oppositeDirIllegal" or something.

Alexgamer470 replied:
You can't check in an if statement if a boolean is true. That's why I wrote it that way and that's also why the bool doesn't have normal name like oppDir. And I think noOppDir is a better name than yours though and that we could just leave it as it is till we find a better one.

Edit: this is wrong

staxfur replied:
so first, an if statement works with boolean logic so that will work and seconds that implementation hassome problems, like you can't go backwards if snakeLength = 0, plus a bug, if you go up and press left+ down you actually go down, so it probably would be better to move the check into the actuall movement, or into a new function, that gets called by update() between input() and the snake movement function.

Alexgamer470 replied:
i know these problems but this branch is just for the on/off switch and not bugfixing, because this problem has existed before.

staxfur replied:
And your var name doesn't makes clear what its about -> as well as using acronyms is most of the time bad practise. Longer names are better than short ones sometimes.

Alexgamer470 committed:
last touches/fixes and made requested changes from pull request

better writing of the comment and code, renamed the bool, fixed a bug and you can now always move in the opposite direction as long as your score is equal to zero.
The feature is now successfully implemented.

Commit:
small cleanup

MCWertGaming commented:
good addition! Pull-requestis approved and merged!

Cheers!
